src/app/services/explorer/explorer.service.ts

```diff
@@ -54,24 +57,34 @@ export class ExplorerService {
     return this.internalMaxDecimals;
   }
 
-  /**
-   * Lets know if the initialize() function has already been called.
-   */
-  private initialized = false;
+  private nodeUrlSubscription: Subscription;
+  private initializationSubscription: Subscription;
 
   constructor(
     private api: ApiService,
   ) { }
 
+  /**
+   * Initializes the service.
+   */
   initialize() {
-    if (this.initialized) {
-      return;
-    }
-    this.initialized = true;
+    namedAddresses.forEach(namedAddress => {
+      this.namedAddressesMap.set(namedAddress.address, namedAddress.name);
+    });
+
+    this.getNodeInfo(0);
+  }
 
-    // Get basic information about the node.
-    this.api.getHealth().subscribe(response => {
+  /**
+   * Gets the basic info about the backend.
+   * @param delayMs Delay before starting to get the data.
+   */
+  private getNodeInfo(delayMs: number) {
+    if (this.initializationSubscription) {
+      this.initializationSubscription.unsubscribe();
+    }
 
+    this.initializationSubscription = of(0).pipe(delay(delayMs), mergeMap(() => this.api.getHealth())).subscribe(response => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/skycoin/skycoin-explorer/pullRequest?prid=7163626)
```suggestion
    this.initializationSubscription = of(0).pipe(delay(delayMs), mergeMap(() => this.api.getHealth())).subscribe((response) => {
```